presto-main/src/test/java/io/prestosql/execution/TestGrantTask.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.execution;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.Session;
+import io.prestosql.connector.CatalogName;
+import io.prestosql.connector.MockConnectorFactory;
+import io.prestosql.metadata.Catalog;
+import io.prestosql.metadata.CatalogManager;
+import io.prestosql.metadata.MetadataManager;
+import io.prestosql.security.AccessControl;
+import io.prestosql.security.AllowAllAccessControl;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.connector.Connector;
+import io.prestosql.sql.analyzer.FeaturesConfig;
+import io.prestosql.sql.tree.Grant;
+import io.prestosql.sql.tree.Identifier;
+import io.prestosql.sql.tree.PrincipalSpecification;
+import io.prestosql.sql.tree.QualifiedName;
+import io.prestosql.transaction.TransactionManager;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.prestosql.connector.CatalogName.createInformationSchemaCatalogName;
+import static io.prestosql.connector.CatalogName.createSystemTablesCatalogName;
+import static io.prestosql.metadata.MetadataManager.createTestMetadataManager;
+import static io.prestosql.sql.tree.PrincipalSpecification.Type.ROLE;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static io.prestosql.transaction.InMemoryTransactionManager.createTestTransactionManager;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestGrantTask
+{
+    @Test
+    public void testGrantToNonExistingRole()
```

Comment:
Would it be possible to have a positive test here?

Reply:
You asked so nicely that I can't resist.